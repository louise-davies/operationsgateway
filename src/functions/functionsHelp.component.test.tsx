import { render } from '@testing-library/react';
import functionsTokensJson from '../mocks/functionTokens.json';
import FunctionsHelp, { FunctionsHelpProps } from './functionsHelp.component';
describe('FunctionsDialog', () => {
  let props: FunctionsHelpProps;

  const createView = () => {
    return render(<FunctionsHelp {...props} />);
  };

  beforeEach(() => {
    props = {
      data: functionsTokensJson,
    };
  });

  afterEach(() => {
    vi.clearAllMocks();
  });

  it('renders help page correctly', async () => {
    const { asFragment } = createView();

    expect(asFragment()).toMatchSnapshot();
  });
});
