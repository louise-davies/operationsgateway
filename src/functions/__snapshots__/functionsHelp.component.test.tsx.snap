// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FunctionsDialog > renders help page correctly 1`] = `
<DocumentFragment>
  <h3
    class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-um3j9i-MuiTypography-root"
  >
    Functions help
  </h3>
  <p
    class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-ejgmvm-MuiTypography-root"
  >
    In the box, start typing data channel names, numbers, mathematical functions, mathematical symbols such as + and integrate. The Wizard will suggest suitable options and indicate using a grey box when each item has been recognised.
  </p>
  <h3
    class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-um3j9i-MuiTypography-root"
  >
    Functions and operators included 
  </h3>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
  >
    <table
      class="MuiTable-root css-rqglhn-MuiTable-root"
    >
      <tbody
        class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            Mathematical Symbols
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            <div
              class="MuiBox-root css-k008qs"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
              >
                +
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
              >
                -
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
              >
                *
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
              >
                /
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
              >
                **
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
              >
                (
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
              >
                )
              </p>
            </div>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            Statistical Functions
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            <div
              class="MuiBox-root css-k008qs"
            >
              <p
                aria-label="Calculate the mean of a trace (using the y variable) or image input. No effect on a scalar input. Implementation: https://numpy.org/doc/stable/reference/generated/numpy.mean.html"
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
                data-mui-internal-clone-element="true"
              >
                mean
              </p>
              <p
                aria-label="Calculate the maximum value in a trace (using the y variable) or image input. No effect on a scalar input. Implementation: https://numpy.org/doc/stable/reference/generated/numpy.max.html"
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
                data-mui-internal-clone-element="true"
              >
                max
              </p>
              <p
                aria-label="Calculate the minimum value in a trace (using the y variable) or image input. No effect on a scalar input. Implementation: https://numpy.org/doc/stable/reference/generated/numpy.min.html"
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
                data-mui-internal-clone-element="true"
              >
                min
              </p>
            </div>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            Exponential And Logarithmic Functions
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            <div
              class="MuiBox-root css-k008qs"
            >
              <p
                aria-label="Raise \`e\` to the power of the input argument (element-wise if a trace or image is provided). Implementation: https://numpy.org/doc/stable/reference/generated/numpy.exp.html"
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
                data-mui-internal-clone-element="true"
              >
                exp
              </p>
              <p
                aria-label="Calculate the logarithm in base \`e\` of the input argument (element-wise if a trace or image is provided). Implementation:https://numpy.org/doc/stable/reference/generated/numpy.log.html"
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
                data-mui-internal-clone-element="true"
              >
                log
              </p>
            </div>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            Waveform Analysis
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            <div
              class="MuiBox-root css-k008qs"
            >
              <p
                aria-label="Calculate the background of a waveform or image. Errors if scalar provided. Implementation (waveform): First, applies smoothing by taking weighted nearest and next-nearest neighbour contributions to y values whose difference from their neighbours is more than 0.2 times the total range in y. The first 25 and last 25 y values in the signal are averaged to give an estimate of the background.Implementation (image): The average pixel value in the 10 by 10 region in the top left of the image is returned."
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
                data-mui-internal-clone-element="true"
              >
                background
              </p>
              <p
                aria-label="Calculate the centre of a waveform. Errors if image or scalar provided. Implementation: First, applies smoothing by taking weighted nearest and next-nearest neighbour contributions to  y values whose difference from their neighbours is more than 0.2 times the total range in y. The maximum y value is then identified along with the x positions bounding the FWHM, and x value at the centre of this region is returned. Note this will not be the same as the x value of the maximum, unless the data is symmetric."
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
                data-mui-internal-clone-element="true"
              >
                centre
              </p>
              <p
                aria-label="Calculate the falling edge of a waveform. Errors if image or scalar provided. Implementation: First, applies smoothing by taking weighted nearest and next-nearest neighbour contributions to  y values whose difference from their neighbours is more than 0.2 times the total range in y. The maximum y value is then identified along with the x positions bounding the FWHM and the upper x value is returned."
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
                data-mui-internal-clone-element="true"
              >
                falling
              </p>
              <p
                aria-label="Calculate the FWHM of a waveform. Errors if image or scalar provided. Implementation: First, applies smoothing by taking weighted nearest and next-nearest neighbour contributions to  y values whose difference from their neighbours is more than 0.2 times the total range in y. The maximum y value is then identified along with the x positions bounding the FWHM and the difference in x value is returned."
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
                data-mui-internal-clone-element="true"
              >
                fwhm
              </p>
              <p
                aria-label="Integrate a waveform or image. Errors if scalar provided. Implementation (waveform): First, applies smoothing by taking weighted nearest and next-nearest neighbour contributions to y values whose difference from their neighbours is more than 0.2 times the total range in y. The y values are then integrated with respect to x, with the last y value neglected.Implementation (image): Return the sum of all pixel values."
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
                data-mui-internal-clone-element="true"
              >
                integrate
              </p>
              <p
                aria-label="Calculate the rising edge of a waveform. Errors if image or scalar provided. Implementation: First, applies smoothing by taking weighted nearest and next-nearest neighbour contributions to  y values whose difference from their neighbours is more than 0.2 times the total range in y. The maximum y value is then identified along with the x positions bounding the FWHM and the lower x value is returned."
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
                data-mui-internal-clone-element="true"
              >
                rising
              </p>
            </div>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            Image Analysis
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            <div
              class="MuiBox-root css-k008qs"
            >
              <p
                aria-label="Calculate the x position of the centroid of an image. Errors if waveform or scalar provided. Implementation: Returns the x position of the image's centroid in terms of pixel values."
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
                data-mui-internal-clone-element="true"
              >
                centroid_x
              </p>
              <p
                aria-label="Calculate the y position of the centroid of an image. Errors if waveform or scalar provided. Implementation: Returns the y position of the image's centroid in terms of pixel values."
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
                data-mui-internal-clone-element="true"
              >
                centroid_y
              </p>
              <p
                aria-label="Calculate the FWHM of an image across the x axis, at the y position of the image centroid. Errors if waveform or scalar provided. Implementation: First, calculates the centroid and extracts the y position. Then applies smoothing by taking weighted nearest and next-nearest neighbour contributions to pixel values whose difference from their neighbours is more than 0.2 times the total range in values. The maximum value is then identified along with the positions bounding the FWHM and the distance in pixels returned."
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
                data-mui-internal-clone-element="true"
              >
                fmhw_x
              </p>
              <p
                aria-label="Calculate the FWHM of an image across the y axis, at the x position of the image centroid. Errors if waveform or scalar provided. Implementation: First, calculates the centroid and extracts the x position. Then applies smoothing by taking weighted nearest and next-nearest neighbour contributions to pixel values whose difference from their neighbours is more than 0.2 times the total range in values. The maximum value is then identified along with the positions bounding the FWHM and the distance in pixels returned."
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
                data-mui-internal-clone-element="true"
              >
                fmhw_y
              </p>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`FunctionsDialog renders help page correctly 1`] = `
<DocumentFragment>
  <h3
    class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-um3j9i-MuiTypography-root"
  >
    Functions help
  </h3>
  <p
    class="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom css-ejgmvm-MuiTypography-root"
  >
    In the box, start typing data channel names, numbers, mathematical functions, mathematical symbols such as + and integrate. The Wizard will suggest suitable options and indicate using a grey box when each item has been recognised.
  </p>
  <h3
    class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-um3j9i-MuiTypography-root"
  >
    Functions and operators included 
  </h3>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
  >
    <table
      class="MuiTable-root css-rqglhn-MuiTable-root"
    >
      <tbody
        class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            Mathematical Symbols
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            <div
              class="MuiBox-root css-k008qs"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
              >
                +
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
              >
                -
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
              >
                *
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
              >
                /
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
              >
                **
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
              >
                (
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
              >
                )
              </p>
            </div>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            Statistical Functions
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            <div
              class="MuiBox-root css-k008qs"
            >
              <p
                aria-label="Calculate the mean of a trace (using the y variable) or image input. No effect on a scalar input. Implementation: https://numpy.org/doc/stable/reference/generated/numpy.mean.html"
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
                data-mui-internal-clone-element="true"
              >
                mean
              </p>
              <p
                aria-label="Calculate the maximum value in a trace (using the y variable) or image input. No effect on a scalar input. Implementation: https://numpy.org/doc/stable/reference/generated/numpy.max.html"
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
                data-mui-internal-clone-element="true"
              >
                max
              </p>
              <p
                aria-label="Calculate the minimum value in a trace (using the y variable) or image input. No effect on a scalar input. Implementation: https://numpy.org/doc/stable/reference/generated/numpy.min.html"
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
                data-mui-internal-clone-element="true"
              >
                min
              </p>
            </div>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            Exponential And Logarithmic Functions
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            <div
              class="MuiBox-root css-k008qs"
            >
              <p
                aria-label="Raise \`e\` to the power of the input argument (element-wise if a trace or image is provided). Implementation: https://numpy.org/doc/stable/reference/generated/numpy.exp.html"
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
                data-mui-internal-clone-element="true"
              >
                exp
              </p>
              <p
                aria-label="Calculate the logarithm in base \`e\` of the input argument (element-wise if a trace or image is provided). Implementation:https://numpy.org/doc/stable/reference/generated/numpy.log.html"
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
                data-mui-internal-clone-element="true"
              >
                log
              </p>
            </div>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            Waveform Analysis
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            <div
              class="MuiBox-root css-k008qs"
            >
              <p
                aria-label="Calculate the background of a waveform or image. Errors if scalar provided. Implementation (waveform): First, applies smoothing by taking weighted nearest and next-nearest neighbour contributions to y values whose difference from their neighbours is more than 0.2 times the total range in y. The first 25 and last 25 y values in the signal are averaged to give an estimate of the background.Implementation (image): The average pixel value in the 10 by 10 region in the top left of the image is returned."
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
                data-mui-internal-clone-element="true"
              >
                background
              </p>
              <p
                aria-label="Calculate the centre of a waveform. Errors if image or scalar provided. Implementation: First, applies smoothing by taking weighted nearest and next-nearest neighbour contributions to  y values whose difference from their neighbours is more than 0.2 times the total range in y. The maximum y value is then identified along with the x positions bounding the FWHM, and x value at the centre of this region is returned. Note this will not be the same as the x value of the maximum, unless the data is symmetric."
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
                data-mui-internal-clone-element="true"
              >
                centre
              </p>
              <p
                aria-label="Calculate the falling edge of a waveform. Errors if image or scalar provided. Implementation: First, applies smoothing by taking weighted nearest and next-nearest neighbour contributions to  y values whose difference from their neighbours is more than 0.2 times the total range in y. The maximum y value is then identified along with the x positions bounding the FWHM and the upper x value is returned."
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
                data-mui-internal-clone-element="true"
              >
                falling
              </p>
              <p
                aria-label="Calculate the FWHM of a waveform. Errors if image or scalar provided. Implementation: First, applies smoothing by taking weighted nearest and next-nearest neighbour contributions to  y values whose difference from their neighbours is more than 0.2 times the total range in y. The maximum y value is then identified along with the x positions bounding the FWHM and the difference in x value is returned."
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
                data-mui-internal-clone-element="true"
              >
                fwhm
              </p>
              <p
                aria-label="Integrate a waveform or image. Errors if scalar provided. Implementation (waveform): First, applies smoothing by taking weighted nearest and next-nearest neighbour contributions to y values whose difference from their neighbours is more than 0.2 times the total range in y. The y values are then integrated with respect to x, with the last y value neglected.Implementation (image): Return the sum of all pixel values."
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
                data-mui-internal-clone-element="true"
              >
                integrate
              </p>
              <p
                aria-label="Calculate the rising edge of a waveform. Errors if image or scalar provided. Implementation: First, applies smoothing by taking weighted nearest and next-nearest neighbour contributions to  y values whose difference from their neighbours is more than 0.2 times the total range in y. The maximum y value is then identified along with the x positions bounding the FWHM and the lower x value is returned."
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
                data-mui-internal-clone-element="true"
              >
                rising
              </p>
            </div>
          </td>
        </tr>
        <tr
          class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            Image Analysis
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            <div
              class="MuiBox-root css-k008qs"
            >
              <p
                aria-label="Calculate the x position of the centroid of an image. Errors if waveform or scalar provided. Implementation: Returns the x position of the image's centroid in terms of pixel values."
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
                data-mui-internal-clone-element="true"
              >
                centroid_x
              </p>
              <p
                aria-label="Calculate the y position of the centroid of an image. Errors if waveform or scalar provided. Implementation: Returns the y position of the image's centroid in terms of pixel values."
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
                data-mui-internal-clone-element="true"
              >
                centroid_y
              </p>
              <p
                aria-label="Calculate the FWHM of an image across the x axis, at the y position of the image centroid. Errors if waveform or scalar provided. Implementation: First, calculates the centroid and extracts the y position. Then applies smoothing by taking weighted nearest and next-nearest neighbour contributions to pixel values whose difference from their neighbours is more than 0.2 times the total range in values. The maximum value is then identified along with the positions bounding the FWHM and the distance in pixels returned."
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
                data-mui-internal-clone-element="true"
              >
                fmhw_x
              </p>
              <p
                aria-label="Calculate the FWHM of an image across the y axis, at the x position of the image centroid. Errors if waveform or scalar provided. Implementation: First, calculates the centroid and extracts the x position. Then applies smoothing by taking weighted nearest and next-nearest neighbour contributions to pixel values whose difference from their neighbours is more than 0.2 times the total range in values. The maximum value is then identified along with the positions bounding the FWHM and the distance in pixels returned."
                class="MuiTypography-root MuiTypography-body1 css-o8r201-MuiTypography-root"
                data-mui-internal-clone-element="true"
              >
                fmhw_y
              </p>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
